conseil-schema/src/main/scala/tech/cryptonomic/conseil/schema/GenSchema.scala

```diff
@@ -1,51 +1,147 @@
+    | - classpath fallback: application.conf$classpathProvided
+    |
+    | The source files will be generated under the folder /tmp/conseil-slick
+    |
+    | If you need to customize the database connection defaults, pass the file name as argument.
+    | You're expected to provide a top-level hocon section named ${configSection.drop(1)}
+    | An example looks like
+    |
+    | ${configSection.drop(1)} {
+    |   profile: "slick.jdbc.PostgresProfile$$"
+    |   db {
+    |     driver: "org.portgresql.Driver"
+    |     url: "jdbc:postgresql:<host>/<db-name>"
+    |     user: ""
+    |     password: ""
+    |   }
+    | }
+    |
+    | If passing a custom file definition as argument to the task, remember
+    | you can <include required(classpath("application"))> to have everything
+    | provided by default as per the local docker-compose definitions, and
+    | only override what you need to.
+    """.stripMargin)
 
-  final case class PostgresConfig(databaseName: String, user: String, password: String) extends DatabaseConfig {
-    lazy val url = s"jdbc:postgresql://localhost/${databaseName}"
-    lazy val jdbcDriver = "org.postgresql.Driver"
-    lazy val slickProfile = "slick.jdbc.PostgresProfile"
-    lazy val `package` = "tech.cryptonomic.conseil.common.tezos"
-    lazy val dest = "/tmp/slick"
+    externallyProvided getOrElse classpathProvided
   }
 
-  //applies convention to uses CamelCase when reading config fields
-  implicit def hint[T] = ProductHint[T](ConfigFieldMapping(CamelCase, CamelCase))
-
-  val conseilDbConf = loadConfig[PostgresConfig](namespace = "conseil.db.properties")
-
-  conseilDbConf.foreach { cfg =>
-    println(s"Generating database Tables source file under ${cfg.dest}, in package ${cfg.`package`}")
-    slick.codegen.SourceCodeGenerator.main(
-      Array(
-        cfg.slickProfile,
-        cfg.jdbcDriver,
-        cfg.url,
-        cfg.dest,
-        cfg.`package`,
-        cfg.user,
-        cfg.password
-      )
+  io(uri = confUri, outputDir = None).unsafeRunSync()
+
+  /** Essentially follow along what the [[SourceCodeGenerator.run]] does,
+    * yet selecting only a number of tables, grouping the results in different
+    * source files based on the namespace.
+    *
+    * This keeps the different blockchains' db models separated, if we follow the
+    * namespace rule.
+    *
+    * @param uri will point to the configuration file, possibly in a subsection, using fragments
+    * @param outputDir we actually define this inside the configuration, but we could change that
+    * @return an IO effect to be executed via .unsafeRunSync() or similar calls
+    */
+  def io(
+      uri: URI,
+      outputDir: Option[String]
+  ): IO[Unit] = {
+    implicit val shift = IO.contextShift(ExecutionContext.global)
+
+    //we read configuration values
+    val dc = DatabaseConfig.forURI[PostgresProfile](uri)
+    val basePackage = dc.config.getString("codegen.package")
```

Comment:
Do we want to check this value against nullability? I believe, later in the code we may look for schema from `null.$schema`. I think it's worth to add some `println` here at least, for better user-experience. I don't think that this configuration will be missing in our case, but yet, someone may forgot or remove it for some reason, so please consider.